src/main/java/software/sigma/bu003/internship/vedmid_andrii/service/PartService.java

```diff
@@ -0,0 +1,75 @@
+package software.sigma.bu003.internship.vedmid_andrii.service;
+
+import lombok.AllArgsConstructor;
+import org.springframework.stereotype.Service;
+import software.sigma.bu003.internship.vedmid_andrii.entity.Part;
+import software.sigma.bu003.internship.vedmid_andrii.exception.PartNotFoundException;
+import software.sigma.bu003.internship.vedmid_andrii.repository.PartRepository;
+
+import java.util.List;
+import java.util.Objects;
+
+@Service
+@AllArgsConstructor
+public class PartService {
+
+    private final PartRepository partRepository;
+
+    public Part createPart(Part part) {
+        checkIsPart(part.getBrand(), part.getCode());
+
+        Part partFromDB = partRepository.findByBrandAndCode(part.getBrand(), part.getCode());
+
+        if(!Objects.isNull(partFromDB)){
+            return partFromDB;
+        } else {
+            return partRepository.save(part);
+        }
+    }
+
+    public List<Part> getAllParts() {
+        return partRepository.findAll();
+    }
+
+    public Part getPart(String brand, String code) {
+        Part partFromDB = partRepository.findByBrandAndCode(brand, code);
+
+        checkPartInDB(partFromDB);
+
+        return partFromDB;
+    }
+
+    public Part updatePart(Part part) {
```

Comment:
questin is what you gonna do here, if update then you have to save all Part part and validate in controller that price and description @NotEmpty, if merge then it;s a different story